rtos/Kernel.h

```diff
@@ -0,0 +1,98 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#ifndef KERNEL_H
+#define KERNEL_H
+
+#include <stdint.h>
+#include "mbed_rtos1_types.h"
+
+namespace rtos {
+/** \addtogroup rtos */
+/** @{*/
+
+/** The Kernel class allows controlling and obtaining kernel information.
+ *
+ * The Kernel object cannot be instantiated - all functions are static.
+ */
+class Kernel {
```

Comment:
Why is this a class instead of a namespace ?

Reply:
No good answer to that. I don't have a preference either way.

My aim was merely that these methods appear much the same as all the static `Thread` methods to casual users. A namespace might be a nicer way of doing that.

Reply:
In practice it doesn't change much; namespace or class with only static members are not very different however semantically a namespace indicates that `kernel` is **not** an object and therefore cannot be instantiated as one; using a namespace clarifies the intent. 

It also avoid to reference Kernel as a data structure in the documentation generated by doxygen.